### PR TITLE
doc: remove over-explanation

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/libcore/iter.rs [hunk 1]
```diff
@@ -2553,10 +2553,7 @@ pub trait Extend<A> {
     ///
     /// # Examples
     ///
-    /// Basic usage:
```
Comment (steveklabnik) [inline src/libcore/iter.rs]: This is becoming more common throughout the standard library as I go through things, so it should stay.

Comment (tshepang) [inline src/libcore/iter.rs]: Seems not needed since there is no "Advanced usage" example

Comment (steveklabnik) [inline src/libcore/iter.rs]: It's about consistency.

Comment (tshepang) [inline src/libcore/iter.rs]: I would call it consistency if it was everywhere, but it isn't.

Comment (tshepang) [inline src/libcore/iter.rs]: I would also remove all those instances, unless they were accompanied with "Advanced usage" examples.

Comment (steveklabnik) [inline src/libcore/iter.rs]: As I go through the docs, I'm adding them, so this will become gradually so over time.

**Comment by steveklabnik** [comment]
I think I'm :-1: on this PR overall. Extra explanation is good. Docs don't have to be terse.

**Pull request closed without merging**